appinventor/components/src/com/google/appinventor/components/runtime/CloudDB.java

```diff
@@ -557,6 +559,39 @@ public boolean UseSSL() {
     return useSSL;
   }
 
+  /**
+   * Set to `true`{:.logic.block} to have an UpdateDone event triggered when
+   * opertions that store values (StoreValue, AppendValueToList, ClearTag)
+   * complete. This is set to `false`{:.logic.block} by default because some
+   * existing applications (read: SketchAndGuess) do a lot of stores and
+   * sending an event for each one could easily overload some devices.
+   *
+   * This flag is particularly useful in situations where you are storing
+   * a large object (for example an image) that may take some time and you
+   * want to know when the store is complete.
+   *
+   * @param provideUpdateDone true if UpdateDone events will be signaled
+   *
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_BOOLEAN,
+    defaultValue = "False")
+  @SimpleProperty(userVisible = true)
+  public void ProvideUpdateDone(boolean provideUpdateDone) {
+    if (this.provideUpdateDone != provideUpdateDone) {
+      this.provideUpdateDone = provideUpdateDone;
+    }
+  }
+
+  @SimpleProperty(category = PropertyCategory.BEHAVIOR, userVisible = true,
+    description = "Set to true to cause the UpdateDone event to happen when " +
+      "operations that update CloudDB (StoreValue, AppendValueToList, ClearTag) " +
+      "complete. Special note for StoreValue. If many store values calls are " +
+      "made in quick succession, only one UpdateDone event may be triggered when " +
+      "the last one completes.")
+  public boolean ProvideUpdateDone() {
+    return provideUpdateDone;
+  }
+
```

Comment:
What is the motivation for making this a user-controllable property? Nobody will have the event block in their projects (since it's new), so turning on the behavior should be a matter of adding the block.

Reply:
I imagine that people might be confused dragging in the block but not having the event fire because the property defaults to false.

Reply:
I'm trying to avoid the overhead of calling EventDispatch.dispatchEvent. Let me look over that code and see what the real cost might be. SketchAndGuess creates a *lot* events.